tinygrad/codegen/linearizer.py

```diff
@@ -225,78 +225,65 @@ def end_loop(xx:List[Variable]):
     if_gate: Optional[UOp] = None
 
     # reduce op
-    fake_reduce_idxs = []
+    fake_reduce_idxs: List[Variable] = []
+    acc_idxs = global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs
     if self.reduceop is not None:
       # define indexes
       reduce_idxs = [Variable(f"ridx{i}", 0, self.full_shape[i]-1) for i in range(self.first_reduce+len(self.group_for_reduce), self.shape_len-self.upcasted)]
       fake_reduce_idxs = [x*0 for x in reduce_idxs]
+      acc_idxs = global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs
```

Comment:
Why is this assigned a second time?

Reply:
I removed the top assignment and added to the else on the reduceop below.